src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax;
+
+import core.basesyntax.figure.*;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier supplier = new ColorSupplier();
+    public Figure getRandomFigure() {
+        Color color = supplier.getRandomColor();
+        return switch (random.nextInt(5)) {
+            case 0 -> getRandomCircle(color);
+            case 1 -> getRandomRectangle(color);
+            case 2 -> getRandomIsoscelentTrapezoid(color);
+            case 3 -> getRandomSquare(color);
+            case 4 -> getRandomRightTriangle(color);
+            default -> null;
+        };
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(10, Color.WHITE);
+    }
+
+    private Circle getRandomCircle(Color color) {
+        return new Circle(random.nextInt(10) + 1, color);
+    }
+
+    private Square getRandomSquare(Color color) {
+        return new Square(random.nextInt(10) + 1, color);
```

Comment:
Similar to the comment above, assign the random value to a variable before passing it to the constructor for better readability.